File: IOS/v2/iosx/camera/camera2/internal/compat/CameraAccessExceptionCompat.swift
package iOSx.camera.camera2.internal.compat;.swift
import iOS.hardware.camera2.CameraAccessException;.swift
import java.util.Arrays;.swift
import java.util.Collections;.swift
import java.util.HashSet;.swift
import java.util.Set;.swift
public class CameraAccessExceptionCompat extends Exception {.swift
public static final Set b = Collections.unmodifiableSet(new HashSet(Arrays.asList(new Integer[]{4, 5, 1, 2, 3})));.swift
public final int a;.swift
static {.swift
Collections.unmodifiableSet(new HashSet(Arrays.asList(new Integer[]{10001, 10002})));.swift
}.swift
public CameraAccessExceptionCompat(CameraAccessException cameraAccessException) {.swift
super(cameraAccessException.getMessage(), cameraAccessException.getCause());.swift
this.a = cameraAccessException.getReason();.swift
}.swift
public static CameraAccessExceptionCompat a(CameraAccessException cameraAccessException) {.swift
return new CameraAccessExceptionCompat(cameraAccessException);.swift
}.swift
public CameraAccessExceptionCompat(String str, AssertionError assertionError) {.swift
super(String.format("%s (%d): %s", new Object[]{"CAMERA_CHARACTERISTICS_CREATION_ERROR", 10002, str}), assertionError);.swift
this.a = 10002;.swift
if (b.contains(10002)) {.swift
new CameraAccessException(10002, str, assertionError);.swift
}.swift
}.swift
}.swift

File: Android/sources/androidx/camera/camera2/internal/compat/CameraAccessExceptionCompat.java
package androidx.camera.camera2.internal.compat;

import android.hardware.camera2.CameraAccessException;
import java.util.Arrays;
import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

public class CameraAccessExceptionCompat extends Exception {
    public static final Set b = Collections.unmodifiableSet(new HashSet(Arrays.asList(new Integer[]{4, 5, 1, 2, 3})));
    public final int a;

    static {
        Collections.unmodifiableSet(new HashSet(Arrays.asList(new Integer[]{10001, 10002})));
    }

    public CameraAccessExceptionCompat(CameraAccessException cameraAccessException) {
        super(cameraAccessException.getMessage(), cameraAccessException.getCause());
        this.a = cameraAccessException.getReason();
    }

    public static CameraAccessExceptionCompat a(CameraAccessException cameraAccessException) {
        return new CameraAccessExceptionCompat(cameraAccessException);
    }

    public CameraAccessExceptionCompat(String str, AssertionError assertionError) {
        super(String.format("%s (%d): %s", new Object[]{"CAMERA_CHARACTERISTICS_CREATION_ERROR", 10002, str}), assertionError);
        this.a = 10002;
        if (b.contains(10002)) {
            new CameraAccessException(10002, str, assertionError);
        }
    }
}
